src/view/components/customerIdWrapper.jsx

```diff
@@ -0,0 +1,200 @@
+import Delete from "@react/react-spectrum/Icon/Delete";
+import ModalTrigger from "@react/react-spectrum/ModalTrigger";
+import Dialog from "@react/react-spectrum/Dialog";
+import "@react/react-spectrum/Form"; // needed for spectrum form styles
+import PropTypes from "prop-types";
+import WrappedField from "./wrappedField";
+import authenticatedStateOptions from "../constants/authenticatedStateOptions";
+import getDefaultCustomerId from "../utils/getDefaultCustomerId";
+import ErrorMessage from "./errorMessage";
+import InfoTipLayout from "./infoTipLayout";
+
+function CIdWrapper({ values }) {
+  // const [namespaceOptions, setNamespaceOptions] = useState([]);
+  const [showStuff, setShowStuff] = useState({});
+  let returnValue = null;
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        // const nOptions = await loadIdentityNamespaces();
+
+        setShowStuff({
+          value: "content"
+        });
+        // setNamespaceOptions(nOptions);
+      } catch (error) {
+        setShowStuff({
+          value: "error",
+          message: error.message
+        });
+      }
+    }
+
+    fetchData();
+  }, []);
+
+  switch (showStuff.value) {
+    case "content":
+      returnValue = (
+        <React.Fragment>
+          <FieldLabel labelFor="instanceName" label="Instance" />
+          <div>
+            <WrappedField
+              id="instanceName"
+              name="instanceName"
+              component={Textfield}
+              componentClassName="u-fieldLong"
+              supportDataElement
+            />
+          </div>
+          <FieldArray
+            name="customerIds"
+            render={arrayHelpers => {
+              return (
+                <React.Fragment>
+                  <div className="u-gapTop u-alignRight">
+                    <Button
+                      id="addCustomerId"
+                      label="Add Customer ID"
+                      onClick={() => {
+                        arrayHelpers.push(getDefaultCustomerId());
+                      }}
+                    />
+                  </div>
+                  {((Array.isArray(values.customerIds) &&
+                    values.customerIds.length) ||
+                    null) && (
+                    <Heading variant="subtitle2">Customer IDs</Heading>
+                  )}
+                  <div>
+                    {((Array.isArray(values.customerIds) &&
+                      values.customerIds.length) ||
+                      null) &&
+                      values.customerIds.map((customerId, index) => (
+                        <Well key={index}>
+                          <div>
+                            <FieldLabel
+                              labelFor="namespaceField"
+                              label="Namespace"
+                            />
+                            <div>
+                              <WrappedField
+                                id="namespaceField"
```

Comment:
When you have multiple customer IDs, you have multiple elements with the same ID (`namespaceField`) in the same document. By spec, we shouldn't have multiple elements with the same ID. We were able to get away with this pattern on the extension configuration view because only one accordion item is in the document at one time. In this case though, multiple wells are in the document at the same time, resulting in duplicate IDs.

Reply:
Thanks for clarifying. I was wondering about this.